### ISSUE TITLE
autosummary document imported functions

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1061]
Hello,

I'm looking for a way to exclude imported modules and functions.

The attached examples list **isfile** in functions of my module.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1061
- Originally reported by: Anonymous
- Originally created at: 2012-12-18T16:59:40.487

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-01-20 02:12:50+00:00_

I can not reproduce autodocing for imported modules and functions.

Please let me know about your 'autodoc' relative directive using in rst file.

**Comment by shimizukawa** [comment]
_From Tiago Coutinho on 2013-10-10 09:15:55+00:00_

Hello,
I am using sphinx 1.1.3 to document my python project. I can confirm the bug.

I was preparing a bug report when I saw this one. I post my original report with the hope that it can help solve the bug:

I am using autosummary with autosummary_generate = True in the conf.py

The autosummary is generating documentation for functions/classes which are imported from another module.
Example:

```
#!python

# --- module1.py ----
def foo():
    """foo documentation"""
    pass

# --- module2.py ---
from module1 import foo
def bar():
    """bar documentation"""
    foo()
```

and the following RST:

```
#!rst

.. doc.rst

Documentation
============

.. autosummary::
   :toctree:

   module2
```

will generate a RST like:

```
#!rst

module2
=======

.. automodule:: module2

   .. rubric:: Functions

   .. autosummary::

      foo
      bar
```

Please don't hesitate to ask for more info if needed.
Thanks,
Tiago

**Comment by shimizukawa** [comment]
_From [samuele_c](https://bitbucket.org/samuele_c) on 2014-03-05 00:02:35+00:00_

I'm having the exact same issue in Debian stable and Debian testing. Calling

```
.. autosummary:: 
   :toctree: _autosummary
```

includes functions imported from other modules in the summary table

**Comment by shimizukawa** [comment]
_From [mzjn](https://bitbucket.org/mzjn) on 2014-08-23 17:44:07+00:00_

A patch has been suggested at Stack Overflow: http://stackoverflow.com/a/25460763

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-08-29 01:53:58+00:00_

We need a `:imported: True` option for `autosummary` (I think False is better for default).

It's a new feature and a incompatible change.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-12-19 03:41:29+00:00_

Issue #1654 was marked as a duplicate of this issue.

**Comment by shimizukawa** [comment]
_From Luc Saffre on 2014-12-19 04:10:55+00:00_

Yes, they are the same issues.

**Comment by embray** [comment]
Unfortunately the fix for this issue in 3313276 broke some things for us in Astropy, because now the call to `documenter.check_module()` is used indiscriminately, whereas we _want_ to document imported objects (because we want to document objects in the top-level namespace they're imported to, and not in the submodules they are defined in, in many cases.

This should at least be optional.  I like the suggestion @shimizukawa made of an `:imported: True` option.

**Comment by jschueller** [comment]
from what I understood, this change was a backward-incompatible commit preventing imported modules to be included in the toctree.

**Comment by RNAer** [comment]
It seems this problem is still not fixed?

**Comment by shimizukawa** [comment]
PR #1663 closes this issue. @birkenfeld please take a look if you have chance.

**Comment by birkenfeld** [comment]
It looks like 1.3.2 already restored the desired behavior.

**Comment by wholmgren** [comment]
Maybe I'm missing something but I believe that this is still an issue, at least for my configuration. I'm using sphinx 1.4.1 and `autosummary_generate = True`. Here are the links to my [conf.py](https://github.com/wholmgren/pvlib-python/blob/doc-reorg2/docs/sphinx/source/conf.py), [top level autosummary](https://raw.githubusercontent.com/wholmgren/pvlib-python/doc-reorg2/docs/sphinx/source/modules.rst), the [template](https://raw.githubusercontent.com/wholmgren/pvlib-python/doc-reorg2/docs/sphinx/source/_templates/autosummary/module.rst) that includes additional autosummary calls, and a sample [page](http://wholmgren-pvlib-python-new.readthedocs.org/en/doc-reorg2/generated/pvlib.tracking.html) with the unwanted imported documentation (the `sind`, `cosd`, and `Location` are all imported).

The change suggested by this [SO answer](http://stackoverflow.com/questions/25405110/sphinx-autosummary-with-toctree-also-lists-imported-members/25460763#25460763) worked for me.

Thanks in advance, and apologies if I am in fact missing something.

**Comment by tk0miya** [comment]
Reopen to investigation

**Comment by shimizukawa** [comment]
Fixed by #3235.
